src/derivkit/forecasting/fisher.py

```diff
@@ -0,0 +1,255 @@
+"""Fisher forecasting utilities."""
+
+from typing import Any, Callable
+
+import numpy as np
+from numpy.typing import ArrayLike, NDArray
+
+from derivkit.calculus_kit import CalculusKit
+from derivkit.forecasting.forecast_core import get_forecast_tensors
+from derivkit.utils.concurrency import normalize_workers
+from derivkit.utils.linalg import solve_or_pinv
+from derivkit.utils.validate import validate_covariance_matrix
+
+
+def build_fisher_matrix(
+    function: Callable[[ArrayLike], float | NDArray[np.floating]],
+    theta0: ArrayLike,
+    cov: ArrayLike,
+    *,
+    method: str | None = None,
+    n_workers: int = 1,
+    **dk_kwargs: Any,
+) -> NDArray[np.floating]:
+    """Computes the Fisher information matrix for a given model and covariance.
+
+    Args:
+        function: The scalar or vector-valued model function. It should accept
+            a 1D array-like of parameter values and return either a scalar or
+            an array of observables.
+        theta0: Expansion point in parameter space (fiducial parameters).
+        cov: Covariance matrix of the observables. Must be square with shape
+            ``(n_observables, n_observables)``.
+        method: Derivative method name or alias (e.g., ``"adaptive"``, ``"finite"``).
+            If ``None``, the DerivativeKit default is used.
+        n_workers: Number of workers for per-parameter parallelisation. Default is 1 (serial).
+        **dk_kwargs: Additional keyword arguments forwarded to
+            :meth:`DerivativeKit.differentiate`.
+
+    Returns:
+        Fisher matrix with shape ``(n_parameters, n_parameters)``.
+    """
+    fisher = get_forecast_tensors(
+        function=function,
+        theta0=theta0,
+        cov=cov,
+        forecast_order=1,
+        method=method,
+        n_workers=n_workers,
+        **dk_kwargs,
+    )
+    # get_forecast_tensors already returns a NumPy array here, but enforce dtype.
+    return fisher
+
+
+def build_fisher_bias(
+        function: Callable[[ArrayLike], float | NDArray[np.floating]],
+        theta0: ArrayLike,
+        cov: ArrayLike,
+        *,
+        fisher_matrix: NDArray[np.floating],
+        delta_nu: NDArray[np.floating],
+        n_workers: int = 1,
+        method: str | None = None,
+        rcond: float = 1e-12,
+        **dk_kwargs: Any,
+) -> tuple[NDArray[np.floating], NDArray[np.floating]]:
+    r"""Estimates parameter bias using the stored model, expansion point, and covariance.
+
+    This method quantifies how differences between two data sets—typically an
+    "unbiased" or reference data vector and a "biased" one including a given
+    systematic—propagate into parameter biases when interpreted through a Fisher
+    forecast. It evaluates the model response internally and uses it, together
+    with the stored covariance and provided Fisher matrix, to estimate both the
+    bias vector and the resulting shift in parameter values.
+    For more information, see https://arxiv.org/abs/0710.5171.
```

Comment:
I'd slightly rephrase this:
```suggestion
    Quantifies how differences between two data sets
    propagate into parameter biases when interpreted through a Fisher
    forecast. Typical comparisons are between an "unbiased" or reference
    data vector and a "biased" one including a given systematic.
    
    The function evaluates the model response internally and uses it, together
    with the stored covariance and provided Fisher matrix, to estimate both the
    bias vector and the resulting shift in parameter values.
    For more information, see https://arxiv.org/abs/0710.5171.
```
I am not really a fan of scare quotes, as it suggests that it doesn't really describe what's going on. This leave ambiguity about what the object is.

"a given systematic" suggests that the user has to add something. I'd keep this simpler: state that the function compares the (linear) difference in response to a change in data vector.